apps/src/StudioApp.js

```diff
@@ -539,6 +549,24 @@ StudioApp.prototype.init = function(config) {
       startDialogDiv
     );
   }
+  if (!config.readOnlyWorkspace) {
+    this.addChangeHandler(() => {
+      // if the code has changed (other than whitespace at the beginning or end) and the code is running,
+      // we want to show display an alert to tell the user to reset and run their code again. We trim the whitespace
+      // because droplet sometimes adds an extra newline when switching from block to code mode.
+      if (
+        this.isRunning() &&
+        this.editDuringRunAlert === undefined &&
+        this.getCode().trim() !== this.executingCode.trim()
+      ) {
+        this.editDuringRunAlert = this.displayWorkspaceAlert(
+          'warning',
+          <div>{msg.editDuringRunMessage()}</div>,
```

Comment:
note that we shouldn't use JSX syntax outside the context of a JSX file. I'd recommend either making this a standalone component or constructing it directly using [`createElement`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement)